src/main/java/com/yandex/money/api/typeadapters/model/showcase/container/GroupTypeAdapter.java

```diff
@@ -112,27 +110,16 @@ protected JsonElement serializeItem(Component src, JsonSerializationContext cont
     }
 
     @Override
+    @Nullable
     protected Component deserializeItem(JsonElement src, JsonDeserializationContext context) {
-        return context.deserialize(src, ComponentsTypeProvider.getClassOfComponentType(
-                getTypeFromJsonElement(src)));
+        return ListDelegate.deserializeComponent(src, context);
```

Comment:
Why switching to ListDelegate? Why don't use something like this:

final Component.Type type = getTypeFromJsonElement(src);
return type != null ? context.deserialize(src, ComponentsTypeProvider.getClassOfComponentType(type)) : null;